src/i18n/i18n.service.ts

```diff
@@ -125,4 +219,24 @@ export class I18n {
 		}
 		return value;
 	}
+
+	/**
+	 * Helper method that returns an observable from the internal cache based on the path
+	 *
+	 * @param path looks like `"NOTIFICATION.CLOSE_BUTTON"`
+	 */
+	protected getSubject(path: string): Observable<string> {
+		try {
+			// we run this here to validate the path exists before adding it to the translation map
+			const value = this.getValueFromPath(path) as string;
+			if (this.translations.has(path)) {
```

Comment:
Tiny performance improvement would be to use `get` right away and store the result, then check for not undefined to return the result.

Reply:
Given that a `Map` should have a `O(1)` worst case I don't think we'll see a tangible speedup from that. Plus that would involve another assignment and more garbage collection, so if anything it could be slower.

The current way is also very explicit

Reply:
If you feel this strongly about it, keep it.